homeassistant/components/miraie_ac/config_flow.py

```diff
@@ -0,0 +1,80 @@
+"""Config flow for MirAIe AC integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from py_miraie_ac import AuthException, AuthType, MirAIeAPI
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import CONFIG_KEY_PASSWORD, CONFIG_KEY_USER_ID, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONFIG_KEY_USER_ID): str,
+        vol.Required(CONFIG_KEY_PASSWORD): str,
+    }
+)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
+    """Validate the user input allows us to connect."""
+
+    login_id = data[CONFIG_KEY_USER_ID]
+    password = data[CONFIG_KEY_PASSWORD]
+    async with MirAIeAPI(
+        auth_type=AuthType.MOBILE, login_id=login_id, password=password
+    ) as api:
+        try:
+            _LOGGER.debug("Initializing MirAIe API")
+            await api.initialize()
+        except AuthException as ex:
+            _LOGGER.error("Invalid user ID or password")
```

Comment:
Yes, there is an http call to login and the exception is raised if it comes back with an error.

Reply:
But there are many other possible errors right?
The webservice could be totally unresponsive.
It might return a response with status code 500.

Will these also show up as "Invalid user ID or password" to the user?